nameserver/zone.go

```diff
@@ -121,33 +121,37 @@ func (zone *ZoneDb) AddRecord(ident string, name string, ip net.IP) error {
 	return nil
 }
 
-func (zone *ZoneDb) DeleteRecord(ident string, ip net.IP) error {
+func (zone *ZoneDb) DeleteRecords(ident string, name string, ip net.IP) (count int) {
 	zone.mx.Lock()
 	defer zone.mx.Unlock()
-	pred := func(r dbRecord) bool { return r.Ident == ident && r.IP.Equal(ip) }
-	if index := zone.indexOf(pred); index != -1 {
-		zone.recs = append(zone.recs[:index], zone.recs[index+1:]...)
-		return nil
+
+	if name != "" {
+		name = dns.Fqdn(name)
+	}
+
+	match := func(r dbRecord) bool {
+		return (ident == "" || r.Ident == ident) &&
+			(name == "" || r.Name == name) &&
+			(ip.Equal(net.IP{}) || r.IP.Equal(ip))
```

Comment:
Just a suggestion, but we could check for the `(ident, *, *)`, `(ident, name, *)`, `(ident, name, ip)` cases with something like:

```code:Go
match := func(r dbRecord) bool {
return (ident == "" || (r.Ident == ident && (name == "" || (r.Name == name && (ip.IsUnspecified() || r.IP.Equal(ip))))))
}
```